server/src/main/java/org/elasticsearch/action/support/broadcast/unpromotable/TransportBroadcastUnpromotableAction.java

```diff
@@ -61,29 +63,36 @@ protected TransportBroadcastUnpromotableAction(
         transportService.registerRequestHandler(transportUnpromotableAction, executor, requestReader, new UnpromotableTransportHandler());
     }
 
-    protected abstract void unpromotableShardOperation(Task task, Request request, ActionListener<ActionResponse.Empty> listener);
+    protected abstract void unpromotableShardOperation(Task task, Request request, ActionListener<Response> listener);
```

Comment:
I decided that it's enough to have one type of response for the node and global response, but we can easily change that if we think it would be beneficial to have concrete types for node requests/responses.